src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Centipede engine interface."""
+
+from pathlib import Path
+import re
+
+from clusterfuzz._internal.bot.fuzzers import dictionary_manager
+from clusterfuzz._internal.bot.fuzzers import engine_common
+from clusterfuzz._internal.bot.fuzzers import utils as fuzzer_utils
+from clusterfuzz._internal.metrics import logs
+from clusterfuzz._internal.system import environment
+from clusterfuzz._internal.system import new_process
+from clusterfuzz.fuzz import engine
+
+_CLEAN_EXIT_SECS = 10
+_TIMEOUT = 25
+_SERVER_COUNT = 1
+_RSS_LIMIT = 4096
+_RLIMIT_AS = 5120
+_ADDRESS_SPACE_LIMIT = 0
+_DEFAULT_ARGUMENTS = [
+    '--exit_on_crash=1',
+    f'--timeout={_TIMEOUT}',
+    f'--fork_server={_SERVER_COUNT}',
+    f'--rss_limit_mb={_RSS_LIMIT}',
+    f'--address_space_limit_mb={_ADDRESS_SPACE_LIMIT}',
+]
+
+_CRASH_REGEX = re.compile(r'Crash detected, saving input to (.*)')
```

Comment:
This will incorrectly parse the period of the sentence as an extra dot at the end of the input filename (e.g., `testcase.` instead of `testcase`). I will fix it in a separate PR.